### PR TITLE
MRG: removing some vendored dependencies

### PR DESCRIPTION
**Pull request by matthew-brett**
We don't need OrderedDict copy now we don't support Python 2.6.

Depend on six package - it's very stable at this point, and pip works much
better than in the past.

### CODE REVIEW

**Comment by matthew-brett** [comment]
This one now ready for review.

File: nibabel/externals/__init__.py [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # init for externals package
-try:
-    from collections import OrderedDict
-except ImportError:  # < Python 2.7
-    from .ordereddict import OrderedDict
+from collections import OrderedDict
+
+from ..deprecated import ModuleProxy as _ModuleProxy
+six = _ModuleProxy('nibabel.externals.six')
```
Comment (grlee77) [inline nibabel/externals/__init__.py]: As nibabel directly uses six everywhere now, this is just here to support 3rd party code that may still be using nibabels.external.six?

Comment (matthew-brett) [inline nibabel/externals/__init__.py]: @grlee77 - right - this clause is just in case someone is importing six from nibabel.externals .

**Comment by matthew-brett** [comment]
Thanks for the review - made changes - OK to merge?

**Comment by effigies** [comment]
Looks like callable edits didn't make it into your last commit?

**Comment by matthew-brett** [comment]
Oops - thanks - added callable.

**Comment by effigies** [comment]
Good to merge.

**Comment by matthew-brett** [comment]
Thanks for the review.

**Pull request merged**